apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletTransactionsInfinite.ts

```diff
@@ -19,22 +19,22 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 import { useInfiniteQuery } from '@tanstack/react-query'
 import { useCallback, useMemo, useState } from 'react'
 
+import useLimitedWalletAddresses from '@/api/apiDataHooks/utils/useLimitedWalletAddresses'
 import { useFetchWalletLastTransaction } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
 import { walletTransactionsInfiniteQuery } from '@/api/queries/transactionQueries'
 import { useAppSelector } from '@/hooks/redux'
-import { selectAllAddressHashes } from '@/storage/addresses/addressesSelectors'
 
-const useFetchWalletInfiniteTransactions = () => {
+const useFetchWalletTransactionsInfinite = () => {
   const networkId = useAppSelector((s) => s.network.settings.networkId)
-  const allAddressHashes = useAppSelector(selectAllAddressHashes) // TODO: Display warning if > 80 addresses
+  const { addressHashes, isLimited } = useLimitedWalletAddresses()
```

Comment:
Would it make sense to simplify things by having only one `useFetchWalletAddressesses` hook that we would systematically use to get the wallet addresses? It could take care of limiting addresses if needed and sort them by activity by default. Would it make sense?

I'm still confused by the number of hooks that looks the same:
- useLimitedWalletAddresses (which is actually relying on a fetch, so we should probably use a "fetch" prefix here?)
- useFetchWalletAddressesSortedByActivity
- useFetchWalletAddressesActivityTimestampsSorted

Reply:
Let me know if the answer above covers this: https://github.com/alephium/alephium-frontend/pull/864#discussion_r1790267700

In short, we still need a way to extract only the hashes since that's what our components need (why extract them in every component and not DRY our code by extracting them in one place?), and in other cases we need additional information. In some cases we need max 80 addresses (when we fetch txs for example) and in other cases we want all the addresses (addresses page). There are also cases where we don't need to fetch anything and we simply want to check if an address is an internal address for example, so there we only use Redux.